### ISSUE TITLE
bug with convolution computing

### ISSUE THREAD

**Issue by un-knight** [Tencent/ncnn#1717]
I found that convolution computing would get some fault results in some cases. Below is one of example.

## pytorch code
Executing a convolution computation and export onnx model.
```python
import torch
import torch.nn as nn
import torch.onnx

torch.manual_seed(1)

model = nn.Sequential(
    nn.Conv2d(3, 2, kernel_size=(4,5), stride=1, padding=0)
)
x = torch.ones(1, 3, 4, 5)

input_names = ['input']
output_names = ['output']

torch_out = torch.onnx.export(
    model, x, "norm_conv2d.onnx", verbose=True,
    input_names=input_names, output_names=output_names,
    opset_version=11,
    keep_initializers_as_inputs=True,
    do_constant_folding=True
)

out = model(x)
print(out, out.shape)
```
And then using `onnx2ncnn` converts `norm_conv2d.onnx` into `norm_conv2d.param` and `norm_conv2d.bin`.

## ncnn infer code
```cpp
#include <iostream>
#include <vector>

#include "net.h"


int main(void){
    ncnn::Net model;

    model.load_param("../models/norm_conv2d.param");
    model.load_model("../models/norm_conv2d.bin");
    std::cout << "load model done!" << std::endl;

    ncnn::Mat in;
    in.create(5, 4, 3);
    in.fill(1.0f);
    std::cout << "create mat done!" << std::endl;

    ncnn::Extractor ex = model.create_extractor();
    ex.input("input", in);

    ncnn::Mat out;
    ex.extract("output", out);

    std::cout << out.w << out.h << out.c << std::endl;
    for(int i = 0; i < out.w*out.h*out.c; ++i){
        std::cout << out[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}
```
And here is the content of `norm_conv2d.param`:
```
7767517
2 2
Input            input                    0 1 input
Convolution      output                   1 1 input output 0=2 1=5 11=4 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=1 6=120
```

## result comparison 
pytorch output:
```
tensor([[[[0.2423]],

         [[0.3889]]]], grad_fn=<MkldnnConvolutionBackward>)
```

ncnn output:
```
0.242285 4.56599e-41
```
So the result of ncnn in the second channel is wrong which pytorch's output is 0.3889 while ncnn's output is 4.56599e-41.

And my conclusion is that when one dim of input `w` or `h` equal to `kernel_w` or `kernel_h` with respect, then the ncnn's computation result would get an error.

## other information
I use ncnn with commit 3a34b99fe6a5a2ef0119b412b420f3c8ff3f14dd

**Comment by nihui** [comment]
Because the internal memory inside ncnn Mat structure may not be continuous, you shall not print its content as a linear data array.
If you really need a linear array, out = out.reshape(out.w * out.h * out.c); do the trick with an extra memory copy potentially

https://github.com/Tencent/ncnn/wiki#print-mat-content

https://github.com/Tencent/ncnn/wiki/FAQ-ncnn-produce-wrong-result#blob-may-have-channel-gap

**Comment by un-knight** [comment]
@nihui ok, really thanks!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
